src/AppInstallerRepositoryCore/Rest/Schema/1_0/Interface.cpp

```diff
@@ -0,0 +1,202 @@
+        {
+            if (restApiUri.back() == '/')
+            {
+                restApiUri.pop_back();
+            }
+
+            return restApiUri;
+        }
+
+        utility::string_t GetSearchEndpoint(const std::string& restApiUri)
+        {
+            std::string fullSearchAPI = restApiUri;
+            return utility::conversions::to_string_t(fullSearchAPI.append(ManifestSearchPostEndpoint));
+        }
+
+        utility::string_t GetManifestByVersionEndpoint(
+            const std::string& restApiUri, const std::string& packageId, const std::string& version, const std::string& channel)
+        {
+            std::string versionEndpoint = restApiUri;
+            versionEndpoint.append(ManifestByVersionAndChannelGetEndpoint).append(packageId);
+
+            // Add Version Query param
+            if (!version.empty())
+            {
+                versionEndpoint.append("?version=").append(version);
+            }
+
+            // Add Channel Query param
+            if (!version.empty())
+            {
+                versionEndpoint.append("&channel=").append(channel);
+            }
+
+            return utility::conversions::to_string_t(versionEndpoint);
+        }
+
+        bool IsStringWhitespace(const std::string& value)
+        {
+            for (size_t i = 0; i < value.length(); i++)
+            {
+                if (!std::isspace(value[i]))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        std::optional<std::string> GetStringFromJsonStringValue(const web::json::value& value)
+        {
+            if (value.is_null() || !value.is_string())
+            {
+                return {};
+            }
+
+            std::string result = Utility::Trim(utility::conversions::to_utf8string(value.as_string()));
+
+            if (Utility::IsEmptyOrWhitespace(result))
+            {
+                return {};
+            }
+
+            return result;
+        }
+    }
+
+    Interface::Interface(std::string restApi)
+    {
+         m_restApiUri = GetRestAPIBaseUri(std::move(restApi));
+         m_searchEndpoint = GetSearchEndpoint(m_restApiUri);
+    }
+
+    IRestClient::SearchResult Interface::Search(const SearchRequest& request) const
+    {
+        UNREFERENCED_PARAMETER(request);
+        SearchResult result;
+
+        // TODO: Handle continuation token.
+        HttpClientHelper clientHelper(m_searchEndpoint);
+        web::json::value jsonObject = clientHelper.HandlePost(GetSearchBody(request));
+
+        // Parse json and add results to SearchResult.
+        if (jsonObject.is_null())
+        {
+            return result;
+        }
+
+        auto& dataArray = jsonObject.at(GetJsonKeyNameString(Data)).as_array();
+
+        for (auto& manifestItem : dataArray)
+        {
+            std::optional<std::string> packageId = GetStringFromJsonStringValue(manifestItem.at(GetJsonKeyNameString(PackageIdentifier)));

```

Comment:
>packageId [](start = 39, length = 9)

Make sure to revist these fields as some of them are definitely required.
You can add a todo for now since there'll be more error checking, manifest integrity checkto be performed later and they could potentially be in another file.

Reply:
I saw you handled bad cases by skipping them, I think that's probably enough.

---
In reply to: [594722790](https://github.com/microsoft/winget-cli/pull/765#discussion_r594722790) [](ancestors = 594722790)

Reply:
Yes, I will be adding a json parser that will handle the checks in my next PR.

---
In reply to: [594726878](https://github.com/microsoft/winget-cli/pull/765#discussion_r594726878) [](ancestors = 594726878,594722790)